File: exercice2/scala/src/main/scala/fr/devoxx/devops/logs/sql/SparkSQL4.scala
package fr.devoxx.devops.logs.sql

import fr.devoxx.devops.logs.ApacheAccessLog
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SQLContext

/* Top 3 des plages d'IP */
case class SparkSQL4(rdd: RDD[String] , sqlContext: SQLContext) {

  def process: Array[(String, Long)] = {
    val dataFrame = sqlContext.createDataFrame(rdd.map(ApacheAccessLog.parse));
    dataFrame.registerTempTable("ApacheAccessLog");

    Array(("", 0L))
  }
}

File: exercice2/java/src/main/java/fr/devoxx/devops/logs/sql/SparkSQL4.java
package fr.devoxx.devops.logs.sql;

import fr.devoxx.devops.logs.ApacheAccessLog;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;
import scala.Tuple2;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/* Top 3 des plages d'IP */
public class SparkSQL4 extends SparkSQL {

    public List<Tuple2<String, Long>> process(JavaRDD<String> rdd, SQLContext sqlContext) {
        JavaRDD<ApacheAccessLog> accessLogs = rdd.map(ApacheAccessLog::parse);
        configure(sqlContext, accessLogs);

        return new ArrayList<>();
    }

}
